### PR TITLE
1) List of Cryptocurrencies 

### PR DESCRIPTION
**Pull request by jhonnierandrey**
# List of Cryptocurrencies - challenge #1:

This PR includes : 

* Initialization of the React Native application for both iOS and Android.
* Home screen listing Cryptos as in the mockup with hardcoded values.
* Rendering of the hardcoded values follows API data structure for easy implementation.
* Follow theme scheme.
* Main navigation to "Add a Cryptocurrency" screen.
* Navigation back to Home screen.
* Enabling "Add" button when input field is not empty.

Check the [Demo Here](https://drive.google.com/file/d/1TQMH1Rtp0yNcBH6yAFfudoJClwmrBrB5/view?usp=sharing)

### CODE REVIEW

File: App.tsx [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import React, {Fragment} from 'react';
+import AppStack from './src/navigation/AppStack';
+
+const App = () => {
+  return (
+    <Fragment>
```
Comment (FedericoLeiva12) [inline App.tsx]: Fragment is not necessary

File: src/components/AddCoin/addCoinStyles.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    },
+    submitBox: {
+        width: '100%',
+        alignItems: 'flex-end'
+    },
+    btnAddCoin: {
+        marginVertical: 20,
+        backgroundColor: palette.yellow,
+        paddingVertical: 15,
+        // marginHorizontal: 30,
```
Comment (FedericoLeiva12) [inline src/components/AddCoin/addCoinStyles.ts]: delete comments

File: src/components/AddCoin/addCoinStyles.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        width: '50%',
+        borderRadius: 5,
+    },
+    btnAddCointText: {
+        textAlign: 'center',
+        fontWeight: '700',
+        fontSize: 18,
+        color: palette.lightYellow,
+    },
+    // btnStatusDisabled:{
```
Comment (FedericoLeiva12) [inline src/components/AddCoin/addCoinStyles.ts]: delete comments

File: src/components/Home/Home.tsx [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+                    <Header />
+                    <View style={homeStyles.listContainer}>
+                        <FlatList
+                            data={selectedCryptos}
+                            keyExtractor={item => item.id}
+                            renderItem={({ item }) => <ListItem item={item} />}
+                        />
+                    </View>
+                    <View style={homeStyles.addView}>
+                        <TouchableOpacity onPress={() => { navigation.navigate('AddCoin') }}>
```
Comment (FedericoLeiva12) [inline src/components/Home/Home.tsx]: delete brackets in the function

File: src/components/ListItem/ListItem.tsx [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+              style={listItemStyles.healthImg}
+              source={item?.market_data?.percent_change_usd_last_24_hours > 0 ? buyArrow : sellArrow}
+            />
+            <Text
+              style={
+                item?.market_data?.percent_change_usd_last_24_hours > 0
+                  ? listItemStyles.healthBuy
+                  : listItemStyles.healthSell
+              }>
+              {' '}
```
Comment (FedericoLeiva12) [inline src/components/ListItem/ListItem.tsx]: Is this necessary?

**Review by FedericoLeiva12 — CHANGES_REQUESTED** [review]

File: App.tsx (hunk 1 above)
Comment (jhonnierandrey) [reply App.tsx]: Fragment element  and imported removed.

File: src/components/AddCoin/addCoinStyles.ts (hunk 2 above)
Comment (jhonnierandrey) [reply src/components/AddCoin/addCoinStyles.ts]: Comments removed.

File: src/components/AddCoin/addCoinStyles.ts (hunk 3 above)
Comment (jhonnierandrey) [reply src/components/AddCoin/addCoinStyles.ts]: Comments removed.

File: src/components/Home/Home.tsx (hunk 4 above)
Comment (jhonnierandrey) [reply src/components/Home/Home.tsx]: Brackets removed.

File: src/components/ListItem/ListItem.tsx (hunk 5 above)
Comment (jhonnierandrey) [reply src/components/ListItem/ListItem.tsx]: Not longer needed, I already added some padding with the styles, `{' '}` removed.

File: App.tsx [hunk 6]
```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+import AppStack from './src/navigation/AppStack';
+
+const App = () => {
```
Comment (tomasdematos) [inline App.tsx:4]: use implicit return 
() => (<....)

File: src/components/AddCoin/AddCoin.tsx [hunk 7]
```diff
@@ -0,0 +1,38 @@
+import React, { useState } from 'react';
+import { View, Text, SafeAreaView, TextInput, Pressable, TouchableOpacity } from 'react-native';
+import { addCoinStyles } from './addCoinStyles';
+
+const AddCoin = ({ navigation }) => {
+    const [coinTicker, setCoinTicker] = useState<string>('');
```
Comment (tomasdematos) [inline src/components/AddCoin/AddCoin.tsx:6]: type declaration unnecessary:

useState('')

if you declare a initial value, the state will have the type of the default value already

File: src/components/AddCoin/AddCoin.tsx [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+            <View style={addCoinStyles.main}>
+                <TouchableOpacity onPress={() => { navigation.navigate('Home') }}>
+                    <Text style={addCoinStyles.navigationText}>{'<'} Back to list</Text>
+                </TouchableOpacity>
+                <View style={addCoinStyles.formBox}>
+                    <Text style={addCoinStyles.label}>Add a Cryptocurrency</Text>
+                    <TextInput
+                        style={addCoinStyles.input}
+                        placeholder={'Use a name or ticker symbol...'}
+                        placeholderTextColor={'#666666'}
```
Comment (tomasdematos) [inline src/components/AddCoin/AddCoin.tsx:19]: use the palette

File: src/components/AddCoin/AddCoin.tsx [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+        <SafeAreaView style={addCoinStyles.container}>
+            <View style={addCoinStyles.main}>
+                <TouchableOpacity onPress={() => { navigation.navigate('Home') }}>
+                    <Text style={addCoinStyles.navigationText}>{'<'} Back to list</Text>
+                </TouchableOpacity>
+                <View style={addCoinStyles.formBox}>
+                    <Text style={addCoinStyles.label}>Add a Cryptocurrency</Text>
+                    <TextInput
+                        style={addCoinStyles.input}
+                        placeholder={'Use a name or ticker symbol...'}
```
Comment (tomasdematos) [inline src/components/AddCoin/AddCoin.tsx:18]: unnecessary { }

use: placeholder="Use a name...."

File: src/components/Header/Header.tsx [hunk 10]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import {View, Image, Text} from 'react-native';
+import {headerStyles} from './headerStyles';
+
+import avatar from '../../assets/img/avatar.png';
+
+const Header = () => {
```
Comment (tomasdematos) [inline src/components/Header/Header.tsx:7]: use implicit return

**Comment by jhonnierandrey** [comment]
Not longer needed.

**Pull request closed without merging**